### PR TITLE
fix searching and pagination bugs for topics and consumers page

### PR DESCRIPTION
**Pull request by ajaypratap003**
Fixed below bugs
1. Fix consumer groups pagination issue and breaking page issue
2. Fix Consumer groups for topic pagination issue 
3. Fix consumer groups pagination issue on topic details page 
4.  Fix consumer groups pagination issue on topic edit page 
5. Fix searching issue in consumers groups page on second page due to page number in payload 
6. Fix searching issue in topics page on second page due to page number in payload  
7. Fix consumer groups API issue

### CODE REVIEW

**Review by suyash-naithani — APPROVED** [review]

**Comment by suyash-naithani** [comment]
This looks good @ajaypratap003 !
Remember to make a JIRA and assign it to yourself. We have two existing issues in JIRA for consumer groups that this PR fixes. Please tag this PR there and update the JIRA accordingly

File: src/services/consumer-groups.ts [hunk 1]
```diff
@@ -31,8 +29,8 @@ const getConsumerGroups = async (
   );
   const response: AxiosResponse<ConsumerGroupList> =
     await api.getConsumerGroups(
-      offset,
-      limit,
+      undefined,
```
Comment (riccardo-forina) [inline src/services/consumer-groups.ts:32]: Not sure I'm following, doesn't the API support pagination at all?

Comment (ajaypratap003) [reply src/services/consumer-groups.ts:32]: yes, api supports pagination. API doesn't return correct response with `offset`. Same issue I fixed in Topics API which I discussed with Wojciech and he told me not to use offset. With offset API doesn't return `total` in response, it return count which is not correct.
It's working fine.

Comment (wtrocki) [reply src/services/consumer-groups.ts:32]: That is be cause offset limit and count are deprecated fields for all kafka api

https://github.com/bf2fc6cc711aee1a0c2a/kafka-admin-api/blob/main/kafka-admin/.openapi/kafka-admin-rest.yaml#L1612

What we should do is use page and size instead.

Comment (wtrocki) [reply src/services/consumer-groups.ts:32]: If we see any field missing lets create issue so we can fix it on backend

**Review by riccardo-forina — APPROVED** [review]

File: src/services/consumer-groups.ts (hunk 1 above)
Comment (ajaypratap003) [reply src/services/consumer-groups.ts:32]: @wtrocki Thanks for adding details about offset. yes, we are already passing page and size in API call and it's working fine. 
There is no issue for now.

Comment (wtrocki) [reply src/services/consumer-groups.ts:32]: This is about return data.
Parameters:
https://github.com/bf2fc6cc711aee1a0c2a/kafka-admin-api/blob/main/kafka-admin/.openapi/kafka-admin-rest.yaml#L256-L271

 Returned fields to use in pagination:
 https://github.com/bf2fc6cc711aee1a0c2a/kafka-admin-api/blob/main/kafka-admin/.openapi/kafka-admin-rest.yaml#L1189-L1196
 
 Notice that and others are deprecated (likely to be removed soon)

**Comment by wtrocki** [comment]
Since this is merged we can create follow up jira to remove deprecated fields. 
Protip: I think we could add new eslint rule to fail when deprecated code is used:
https://www.npmjs.com/package/eslint-plugin-deprecation

**Pull request merged**